src/sentry/incidents/logic.py

```diff
@@ -317,10 +321,15 @@ def bulk_build_incident_query_params(incidents, start=None, end=None, windowed_s
             params["project_id"] = project_ids
 
         snuba_filter = get_filter(incident.query, params)
+        conditions = resolve_discover_aliases(snuba_filter)[0].conditions
+        if incident.alert_rule:
```

Comment:
Once the snapshot stuff is merged in, I think it'd be safe to assume an incident always has a non-null alert rule. Maybe still wise to keep the check in tough.

Reply:
Yeah I think we can change the code at some point, but we'd have to delete any existing incidents that have no alert rule and make the column not null.